### ISSUE TITLE
Crashes a lot when fidgeting with Find References and UI docking

### ISSUE THREAD

**Issue by haikarainen** [slynch8/10x#1900]
I've uploaded two crashdumps.

First one happened when I was attempting to dock the "Find References" window at the bottom.

Second one happened when I was clicking on a reference in the "Find References" window.

[CrashDumps.zip](https://github.com/slynch8/10x/files/12034038/CrashDumps.zip)

**Comment by slynch8** [comment]
Are you able to get this docking crash consistently? I can't get it to happen. Could you send a screenshot of how you are docking it.

With the second one, was it doing a full parse at the time?

**Comment by haikarainen** [comment]
No sorry these seem very sporadic. The docking crash only happened the first time I ran it (i poked around, opened a workspace, opened a file, hit "Find References", docked it at the bottom and it crashed).

The other Find References crash seem to happen quite often but cant consistently reproduce it. I tried today and managed to get it to crash once but not again. Here is the dump for that new crash (likely identical stack to one of the other crash dumps):  
[10x_crash_2023_7_13_16_18_50.dmp.gzip.zip](https://github.com/slynch8/10x/files/12040404/10x_crash_2023_7_13_16_18_50.dmp.gzip.zip)


For both crashes I've tried  to clear the cache in appdata.

**Comment by haikarainen** [comment]
And to clarify, the find references crash happens when you press a reference in the "Find References" window.

**Comment by slynch8** [comment]
Yes, same callstack. The weird thing is that it's crashing updating the parse colors for the map scroll bar, I guess on the file you are opening.

**Comment by haikarainen** [comment]
https://github.com/slynch8/10x/issues/1689 Is this related?

**Comment by slynch8** [comment]
I think that only happens now if the d3d device resets while updating drivers.

**Comment by slynch8** [comment]
ah, I think I've got it. It's a threading issue. The preview window gets destroyed on another thread.

**Comment by slynch8** [comment]
the find references crash is fixed in (1.0.56)

If you get the docking bug again can you open a new bug

**Comment by haikarainen** [comment]
Thank you for a quick fix!

Just to let you know, I am considering promoting this product to become the recommended IDE for my companies in-house game engine (not yet released to public, but will be), replacing Visual Studio. This is our second game engine that we've made, first one took 8 years to develop and was later acquired by another studio, this engine however will remain our own and become publicly available for other developers via licensing.

We tweeted about 10X here for reference https://twitter.com/PrklInteractive/status/1679809044771401729
